### PR TITLE
Creating another allthethings to eyes test paint glomming

### PR DESCRIPTION
**Pull request by hannahbergam**
Because there were multiple iterations of the paint glomming algorithm, UI testing of the final result was saved until the end of the project. This PR attempts to cover as many edge cases as possible for this algorithm in an eyes test, so that any inadvertent change to the glomming math will result in a test failure.

The PR also sets up testing for the run/stop button, which can be completed once the closeConnection() method produces a console message.

And finally, this work updates the JavalabRun and JavalabFinish id's for the Javalab run button to match the site-wide conventions: runButton and finishButton. This way, all UI testing lines set up in steps.rb (i.e. "And I wait for the page to fully load") will automatically work.

An image of the edge cases included in the Allthethings level:
image

- jira ticket: https://codedotorg.atlassian.net/browse/CSA-622

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/ui/features/javalab/neighborhood.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  When I open my eyes to test "Paint Glomming Shapes"
+  Given I create a levelbuilder named "Simone"
+  And I am on "http://studio.code.org/s/allthethings/lessons/44/levels/7"
+  And I wait for the page to fully load
+  Then I press "runButton"
+  And I wait for 25 seconds
+  And I see no difference for "paint glomming"
+  Then I close my eyes
+
+#Scenario: Stop Button Closes Connection
```
Comment (bethanyaconnor) [inline dashboard/test/ui/features/javalab/neighborhood.feature]: Did you mean to keep this commented out?

File: dashboard/test/ui/features/javalab/neighborhood.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@eyes
+  Feature: NeighborhoodPainting
```
Comment (bethanyaconnor) [inline dashboard/test/ui/features/javalab/neighborhood.feature]: nit: this spacing seems off?

File: dashboard/test/ui/features/javalab/neighborhood.feature (hunk 1 above)
Comment (hannahbergam) [reply dashboard/test/ui/features/javalab/neighborhood.feature]: I did- I explain in the PR description. I can remove it entirely, but figured since I already wrote it, the next person's task might be easier!

File: dashboard/config/scripts_json/allthethings.script_json [hunk 3]
```diff
@@ -4,8 +4,10 @@
     "wrapup_video_id": null,
     "login_required": false,
     "properties": {
-      "lesson_extras_available": true,
-      "hideable_lessons": true
+      "hideable_stages": true,
+      "stage_extras_available": true,
```
Comment (bethanyaconnor) [inline dashboard/config/scripts_json/allthethings.script_json]: nit: I know this diff is noisy but can you remove `hideable_stages` and `stage_extras_available` here? I'm not sure how they wound up back in this file but they're deprecated

File: dashboard/test/ui/features/javalab/neighborhood.feature (hunk 1 above)
Comment (jmkulwik) [reply dashboard/test/ui/features/javalab/neighborhood.feature]: Can you add a task to Jira and link it in this github comment thread to track the remaining work here?

Comment (bethanyaconnor) [reply dashboard/test/ui/features/javalab/neighborhood.feature]: ahh sorry I missed that! I don't have any preference on removing it versus leaving it commented out so commented out seems good to me!

File: dashboard/config/scripts_json/allthethings.script_json (hunk 3 above)
Comment (hannahbergam) [reply dashboard/config/scripts_json/allthethings.script_json]: Yes definitely! I was having a hard time figuring out the merge, and anything that was put in here happened automatically- thank you for the help!

File: dashboard/test/ui/features/javalab/neighborhood.feature (hunk 2 above)
Comment (hannahbergam) [reply dashboard/test/ui/features/javalab/neighborhood.feature]: Fixed! Thanks:)

File: dashboard/config/scripts_json/allthethings.script_json (hunk 3 above)
Comment (hannahbergam) [reply dashboard/config/scripts_json/allthethings.script_json]: Removed those two lines!

**Review by bethanyaconnor — APPROVED** [review]
Looks good to me! Thanks for normalizing those button ids!

File: dashboard/test/ui/features/javalab/neighborhood.feature (hunk 1 above)
Comment (hannahbergam) [reply dashboard/test/ui/features/javalab/neighborhood.feature]: https://codedotorg.atlassian.net/browse/CSA-624 This includes the work to create the console message (which @moneppo mentioned might differ based on mini-app). I will add a comment there pointing the engineer back to this file to finish this test!
